File: backend/src/main/java/com/back/fairytale/global/security/CorsProperties.java
package com.back.fairytale.global.security;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.List;

@ConfigurationProperties(prefix = "app.cors")
@Data
public class CorsProperties {
    private List<String> allowedOrigins;
    private List<String> allowedMethods;
    private List<String> allowedHeaders;
    private boolean allowCredentials;
    private List<String> exposedHeaders;
}

File: backend/src/main/java/com/back/fairytale/global/security/CorsProperties.kt
package com.back.fairytale.global.security

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.cors")
data class CorsProperties(
    var allowedOrigins: List<String> = emptyList(),
    var allowedMethods: List<String> = emptyList(),
    var allowedHeaders: List<String> = emptyList(),
    var allowCredentials: Boolean = false,
    var exposedHeaders: List<String> = emptyList()
)
